### ISSUE TITLE
Example doesn't work: ‘ft_library’ undeclared

### ISSUE THREAD

**Issue by xobs** [harfbuzz/harfbuzz#2106]
I'm trying to understand how to use harfbuzz, and I understand the most important function is `hb_shape`.  As such, I'm going through the documentation trying to understand what the parameters mean, and I've come across the "simple shaping example" from https://harfbuzz.github.io/ch03s03.html

This describes how to allocate a buffer, set its direction, load a font, and perform shaping.

However, the example does not work.

I have installed both harfbuzz and freetype2, and I'm compiling with `gcc $(pkg-config freetype2 harfbuzz --libs --cflags) test.c -o test`.

All is well up until Step 3: "Create a face and a font, using FreeType for now. ".  This is where things start to go wrong.

It has the following code:

```cpp
      #include <hb-ft.h>
      FT_New_Face(ft_library, font_path, index, &face);
      FT_Set_Char_Size(face, 0, 1000, 0, 0);
      hb_font_t *font = hb_ft_font_create(face);
```

None of these arguments are defined inside the header file.  There is `FT_Library`, which appears to be a typedef, but it doesn't actually work.

The example in the manual should be fixed so that it works.  The introductory sentence " Below is the simplest HarfBuzz shaping example possible." doesn't seem correct, because the example does not appear to contain everything it needs.

**Comment by Gillibald** [comment]
https://www.freetype.org/freetype2/docs/reference/ft2-base_interface.html

**Comment by ebraminio** [comment]
Guess @n8willis can have a look at this but feel free to load the font without freetype using this example https://github.com/harfbuzz/harfbuzz-tutorial/blob/master/hello-harfbuzz-opentype.c

**Comment by xobs** [comment]
@Gillibald I don't see any of those symbols defined there.  The Harfbuzz manual says " Below is the simplest HarfBuzz shaping example possible", however it is incomplete and missing information and is perhaps too simplified.  The example should actually work.

@ebraminio That looks like exactly what I wanted!  In fact, `hello-harfbuzz-freetype`c` seems even better since it it follows what's in the manual.  In fact, the same steps are present.

Now that I've managed to get past that hurdle, I've noticed one other issue with it: The arguments to `hb_ft_font_create()` are incorrect. It takes two arguments, but the "simplest example" only has one argument: `      hb_font_t *font = hb_ft_font_create(face);`
